backend/internal/database/book.go

```diff
@@ -91,3 +94,161 @@ func (db *BookDatabase) FetchBookDetails(ctx context.Context,
 
 	return &book, false, nil
 }
+
+/*
+ * Fetches a book entry from the database given the id
+ */
+func (db *BookDatabase) FetchBook(ctx context.Context, id string) (*models.Book, error) {
+	var book models.Book
+	var query = "SELECT books.id, title, author, image, array_remove(array_agg(lang), NULL) as languages," +
+		"created_at FROM books LEFT JOIN book_contents ON " +
+		"books.id = book_contents.id WHERE books.id = $1 GROUP BY books.id"
+
+	err := db.Conn.QueryRowEx(ctx, query, nil, id).Scan(&book.ID, &book.Title, &book.Author, &book.Image,
+		&book.Languages, &book.CreatedAt)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "error on FetchBook")
+	}
+
+	return &book, nil
+}
+
+/*
+ * Inserts a book into the books table
+ * MUST be called on the first instance a book is inserted into the database
+ * and has no associated language
+ */
+func (db *BookDatabase) InsertBook(ctx context.Context,
+	book models.APICreateBook) (models.Book, error) {
+	var newBook = models.Book{
+		Languages: []string{},
+	}
+	var query string = "INSERT INTO books (title, author, image) " +
+		"VALUES ($1, $2, $3) " +
+		"RETURNING id, title, author, image, created_at"
+	err := db.Conn.QueryRowEx(ctx, query, nil, book.Title, book.Author, book.Image).
+		Scan(&newBook.ID, &newBook.Title, &newBook.Author,
+			&newBook.Image, &newBook.CreatedAt)
+
+	if err != nil {
+		return newBook, errors.Wrap(err, "error on INSERT INTO books in InsertBook")
+	}
+
+	return newBook, nil
+
+}
+
+/*
+ * Inserts a books details into the book_contents table. Returns the
+ * complete books details
+ */
+func (db *BookDatabase) InsertBookDetails(ctx context.Context, id string,
+	book models.APICreateBookContents) (*models.BookDetails, error) {
+	var newBookDetail *models.BookDetails
```

Comment:
Minor thing, any reason you chose to declare this variable manually rather than use the `:=` syntax? Noticed this in a few other places as well

Reply:
I just have a habit of explicitly typing every variable I will use just to be super explicit. I guess thats what happens when Java is your first language lol. I can change it if that's not the standard Go practice

Reply:
Also just because it is a pointer and that's not obvious from the name